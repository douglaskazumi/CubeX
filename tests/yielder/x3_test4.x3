class Spoon() extends Iterable<String>
{
	yielder{
		for(i in input){
			yield complex(i);
		}
	}
	
	fun complex(i:String):String{
		return string(i++i);
	}
}

p := Spoon();
ret := [];

for(a in p){
	for(b in p){
		ret := ret++[a]++[b];
	}
}

return ret;