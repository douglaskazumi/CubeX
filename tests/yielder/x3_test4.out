aa
aa
aa
bb
aa
cc
aa
dd
aa
ee
aa
ff
aa
gg
aa
hh
bb
aa
bb
bb
bb
cc
bb
dd
bb
ee
bb
ff
bb
gg
bb
hh
cc
aa
cc
bb
cc
cc
cc
dd
cc
ee
cc
ff
cc
gg
cc
hh
dd
aa
dd
bb
dd
cc
dd
dd
dd
ee
dd
ff
dd
gg
dd
hh
ee
aa
ee
bb
ee
cc
ee
dd
ee
ee
ee
ff
ee
gg
ee
hh
ff
aa
ff
bb
ff
cc
ff
dd
ff
ee
ff
ff
ff
gg
ff
hh
gg
aa
gg
bb
gg
cc
gg
dd
gg
ee
gg
ff
gg
gg
gg
hh
hh
aa
hh
bb
hh
cc
hh
dd
hh
ee
hh
ff
hh
gg
hh
hh
